pkg/fqdn/dnsproxy/helpers.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsproxy
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/maps/proxymap"
```

Comment:
@aanm I think reducing dependencies on cgo if possible anywhere is a nice feature to have. It can also allow for mocking out of the datapath for easier testing. This can be done as follow-up.